cli/tests/pcluster/utils/test_pcluster_utils.py

```diff
@@ -343,3 +344,40 @@ def _generate_stack_event():
         "StackName": FAKE_STACK_NAME,
         "Timestamp": 0,
     }
+
+
+@pytest.mark.parametrize(
+    "bucket_prefix", ["test", "test-", "prefix-63-characters-long--------------------------------to-cut"]
+)
+def test_generate_random_bucket_name(bucket_prefix):
+    bucket_name = utils.generate_random_bucket_name(bucket_prefix)
+    assert_that(bucket_name).starts_with(bucket_prefix[: 63 - len(bucket_prefix) - 16])
+    assert_that(len(bucket_name)).is_between(3, 63)
```

Comment:
Why not asserting the exact expected length?

Reply:
test added